ONSdigital/eq-survey-runner
# Rabbit mq integration

**What**
Connecting the Survey running to Rabbit MQ and implementing a RabbitMQ health check.

**How to test**
**_Testing eq-submission**_
1. Run the application and set the EQ RABBIT MQ environment variables to valid values (see README)
2. Send a JWT request to one of the following urls:
- http://localhost:5000/jwt
- http://localhost:5000/jwt_signed
- http://localhost:5000/jwt_encrypted

For example:
http://localhost:5000/jwt_signed?token=eyJhbGciOiJSUzI1NiJ9.eyJ1c2VyIjoiamltbXkifQ.oec9V6uxd9nlNsEZud9YIf1Ee0Su-sgVSnBPpR-8AURchaT-L_ulrDSmHRzD0BdEpuaJkbA7DgeXUsuFWpspDgtxZvomv7tOel4yTKHWjZ1BzuWcZhWNMbwwzmmauUyWE7FRMVS1yFFpbpOnvuS6jOJWczVMNQEKp6PEH2Q99ZpJ1k9uKACNzXrILLKPSN3OJHn3sYP5Nn0WGQtol_BByE4V4YrhTSrcy3QUpKwPYcw4FzSchnKab0D1R9GHYHKBQaWf0f9PsbDByzF97UhZSEgJrt-kJ1fu3MCYUyrnCYzNz1W_x22cT5SsEiu34IKEZ2w-9XerafAEMI86FA1MOA
1. Browse to http://<env-name>-rabbitmq1.eq.ons.digital:15672/#/queues and check there is a message on the eq-submission queue.

**_Testing health check**_
1. Browse to the health check and check its reporting rabbit mq available
2. Browse to http://<env-name>-rabbitmq1.eq.ons.digital:15672/#/queues and check there is a message on the eq-test queue.
3. Enter an invalid hostname in EQ_RABBITMQ_URL env value
4. Browse to the health check and check its reporting rabbit mq as unavailable.

**Who can review**
Anyone apart from @warrenbailey
